src/domain/shared/index.types.d.ts

```diff
@@ -29,6 +29,12 @@ type NewDisplayAmount<T extends DisplayCurrency> = {
   displayInMajor: DisplayCurrencyMajorAmount
 }
 
+type PriceAmount<T extends DisplayCurrency> = {
+  priceOfOneSatInMinorUnit: number
```

Comment:
Why are we using `number` for money?

Reply:
It's the one place where `number` is still left I believe. This PR does away with the `DisplayAmount` type (which contained an internal `number`) in favour of the  `NewDisplayAmount` type (which contains internal `bigint`).

We have one property still on `WalletTransaction` type, `displayCurrencyPerSettlementCurrencyUnit`, which is a decimal number that eventually gets converted to an integer `base` and `offset` in the API. I think we can remove this `number` now and convert to `BigInt` in the domain for the `WalletTransaction` type instead of in the API layer. I'd prefer to do that in a follow-up PR though since it's a slightly different concern from what this PR is addressing which is an in-place refactor to swap all instances of `DisplayAmount` for `NewDisplayAmount` which implicitly already removes a lot of `number` values and replaces them with `bigint` ones.

Reply:
> I'd prefer to do that in a follow-up PR though

Done here: https://github.com/GaloyMoney/galoy/pull/2473